test/parallel_api/numeric/numeric.ops/reduce_by_segment.pass.cpp

```diff
@@ -346,13 +346,22 @@ run_test()
 int
 main()
 {
+    // On Windows, we observe incorrect results with this test with a specific compilation order of the
+    // kernels. This is being filed to the compiler team. In the meantime, we can rearrange this test
+    // to resolve the issue on our side.
+#if _PSTL_RED_BY_SEG_WINDOWS_COMPILE_ORDER_BROKEN
```

Comment:
Is it OK here that we wrote this statement without negation?

Reply:
Yes, we want to still keep the test enabled but if the macro is set, then we want to place it above flag predicate tests which works around the issue. This is why we have `#if _PSTL_RED_BY_SEG_WINDOWS_COMPILE_ORDER_BROKEN` here and `#if !_PSTL_RED_BY_SEG_WINDOWS_COMPILE_ORDER_BROKEN` below.

Reply:
ok